e2e/container/kill_test.go

```diff
@@ -0,0 +1,66 @@
+package container
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/cli/e2e/internal/fixtures"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKillContainer(t *testing.T) {
+	containerID := runBackgroundTop(t)
+
+	// Kill with a signal should send the signal to the process
+	// if the signal is not a deadly one, the container should still live on
+	result := icmd.RunCmd(
+		icmd.Command("docker", "kill", "-s", "SIGWINCH", containerID),
+	)
+	result.Assert(t, icmd.Success)
+
+	result = icmd.RunCmd(
+		shell(t, "docker inspect -f '{{ .State.Status }}' %s", containerID),
+	)
+	result.Assert(t, icmd.Success)
+	require.Equal(t, "running", strings.TrimSpace(result.Stdout()))
+
+	// Kill with no signal should kill the process
```

Comment:
Defaults could be tested with unit tests. You could use a SIGTERM here to cover testing the CLI accepts the arg properly.